migrations/20200719172434-change-patients-barcodeValue-to-string.js

```diff
@@ -0,0 +1,33 @@
+"use strict";
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction((t) => {
+      return Promise.all([
+        queryInterface.changeColumn(
+          "patients",
+          "barcodeValue",
+          { type: Sequelize.DataTypes.STRING },
+          {
+            transaction: t,
+          }
+        ),
+      ]);
+    });
+  },
+
+  down: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction((t) => {
+      return Promise.all([
+        queryInterface.changeColumn(
+          "patients",
+          "barcodeValue",
+          { type: 'INTEGER USING CAST("barcodeValue" as INTEGER)' },
+          {
+            transaction: t,
+          }
+        ),
+      ]);
+    });
+  },
```

Comment:
this will set any barcodes that aren't integers to `null` which will ensure the `CAST` will always work; this isn't perfect but considering this down migration shouldn't be run very often, it should be an ok middle ground

**please test this out yourself before committing**
```suggestion
    up: (queryInterface, Sequelize) => {
    return queryInterface.changeColumn(
      "patients", "barcodeValue", { type: Sequelize.DataTypes.STRING }
    );
  },

  down: (queryInterface, Sequelize) => {
    return queryInterface.sequelize.transaction((t) => {
      return Promise.all([
        queryInterface.bulkUpdate('patients',
          { barcodeValue: null },
          { barcodeValue: { [Sequelize.Op.regexp]: '[^0-9]' } },
          { transaction: t }
        ),
        queryInterface.changeColumn(
          "patients",
          "barcodeValue",
          { type: 'INTEGER USING CAST("barcodeValue" as INTEGER)' },
          {
            transaction: t,
          }
        ),
      ]);
    });
  },
```

Reply:
yup tested it, it works! Thanks